src/main/scala/org/apache/spark/streamdm/clusterers/StreamKM.scala

```diff
@@ -80,6 +81,35 @@ class StreamKM extends Clusterer {
     })
   }
   
+   /** 
+   *  Maintain the BucketManager for coreset extraction, given an input DStream of Example.
+   * @param input a stream of instances
+   * @return a DStream of tuples containing the original Example and the
+   * assigned cluster.
+   */
+  def trainStreamKM(input: DStream[Example]):  DStream[(Example, Double)] = {
```

Comment:
It is better to have this implementation inside `assign(input: DStream[Example]): DStream[(Example,Double)]`, then you do not have to change class `ClusteringTrainEvaluate.scala`